pytest_checkdocs/__init__.py

```diff
@@ -37,23 +37,7 @@ def repair_field(raw):
         return textwrap.dedent(' ' * 8 + raw)
 
 
-class CheckdocsItem(pytest.Item, pytest.File):
-    def __init__(self, fspath, parent):
-        # ugly hack to add support for fspath parameter
-        # Ref pytest-dev/pytest#6928
-        super().__init__(fspath, parent)
-
-    @classmethod
-    def from_parent(cls, parent, fspath):
-        """
-        Compatibility shim to support
-        """
-        try:
-            return super().from_parent(parent, fspath=fspath)
-        except AttributeError:
-            # pytest < 5.4
-            return cls(fspath, parent)
-
+class CheckdocsItem(pytest.Item):
```

Comment:
currently in addition to the `CheckdocsItem(pytest.Item)` a  `CheckdocsFile(pytest.File)` is needed to model it correctly

by now i found a good example - https://github.com/asmeurer/pytest-flakes/pull/40/files

Reply:
Hmm. In this case, I'm slightly tempted _not_ to create a `pytest.File` because the subject being collected is a more abstract concept than any particular file (a Python project). It does mean that it has no "file" name in the reporting, but I prefer that to associating the check with a particular project file. I think I'll proceed with this approach for now and consider adding a file later if needed.